src/zope/interface/declarations.py

```diff
@@ -436,8 +436,7 @@ def __call__(self, ob):
 def _implements(name, interfaces, classImplements):
     # This entire approach is invalid under Py3K.  Don't even try to fix
     # the coverage for this block there. :(
-    if PYTHON3: # pragma: no cover
-        raise TypeError('Class advice impossible in Python3')
+    assert not PYTHON3, "Class advice impossible in Python 3"
```

Comment:
The rationale is that `python -O` suppresses assertions:  we don't want to remove that check in that case.